packages/circuits/test/proveReportNegRep.test.ts

```diff
@@ -0,0 +1,196 @@
+import { expect } from 'chai'
+import * as utils from '@unirep/utils'
+import { Identity } from '@semaphore-protocol/identity'
+import { IdentityObject } from './types'
+import { Circuit } from '@unirep/circuits'
+import { defaultProver } from '../provers/defaultProver'
+import crypto from 'crypto'
+const circuit = 'reportNegRepProof'
+
+const genCircuitInput = (config: {
+    reportedEpochKey: any
+    hashUserId: string | bigint
+    reportedEpoch: number | bigint
+    currentEpoch: number | bigint
+    currentNonce: number | bigint
+    chainId: number | bigint
+    attesterId: number | bigint
+}) => {
+    const {
+        reportedEpochKey,
+        hashUserId,
+        reportedEpoch,
+        currentEpoch,
+        currentNonce,
+        chainId,
+        attesterId,
+    } = Object.assign(config)
+
+    const circuitInputs = {
+        reported_epoch_key: reportedEpochKey,
+        hash_user_id: hashUserId,
+        reported_epoch: reportedEpoch,
+        current_epoch: currentEpoch,
+        current_nonce: currentNonce,
+        chain_id: chainId,
+        attester_id: attesterId,
+    }
+    return utils.stringifyBigInts(circuitInputs)
+}
+
+const genProofAndVerify = async (
+    circuit: Circuit | string,
+    circuitInputs: any
+) => {
+    const startTime = new Date().getTime()
+    const { proof, publicSignals } =
+        await defaultProver.genProofAndPublicSignals(circuit, circuitInputs)
+    const endTime = new Date().getTime()
+    console.log(
+        `Gen Proof time: ${endTime - startTime} ms (${Math.floor(
+            (endTime - startTime) / 1000
+        )} s)`
+    )
+    const isValid = await defaultProver.verifyProof(
+        circuit,
+        publicSignals,
+        proof
+    )
+    return { isValid, proof, publicSignals }
+}
+
+const createRandomUserIdentity = (): IdentityObject => {
```

Comment:
Note: We can consider moving these common utils to a separate file.

Reply:
@FoodChain1028 See if you want to make this change in this PR or leave this in the future.

Reply:
I will make it in the future PR :)

Reply:
I just create a issue with this: #410